e2e/registration_test.go

```diff
@@ -150,6 +151,51 @@ func TestE2E_ServiceRegistration_DeregisterWhenStopped(t *testing.T) {
 	assert.False(t, registered)
 }
 
+// TestE2E_ServiceRegistration_RegistrationFailure tests that if CTS is unable
+// to register itself as a service, it will not exit.
+func TestE2E_ServiceRegistration_RegistrationFailure(t *testing.T) {
```

Comment:
Maybe include the test case being tested in the name? 

`TestE2E_ServiceRegistration_ContinueOnRegistrationFailure` or something like that